src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -460,7 +467,11 @@ private void Poll()
             var currentAssignment = _consumer.Assignment.ToImmutableList();
             var initialRebalanceInProcess = _rebalanceInProgress.Value;
 
-            if (_requests.IsEmpty())
+            var partitionsToFetch = _requests.Values.SelectMany(v => v.Topics)
+                .Where(p => currentAssignment.Contains(p))
+                .ToImmutableHashSet();
```

Comment:
Actual fix:

Before, we did not filter the expected fetch partitions against the currently assigned partitions in the `ConsumerClient`, this causes the "Unexpected records" bug.